faker/providers/python/__init__.py

```diff
@@ -32,6 +33,9 @@ def pystr(self, min_chars=None, max_chars=20):
                 ),
             )
 
+    def pystr_format(self, string_format='?#-###{{random_int}}{{random_letter}}', letters=string.ascii_letters):
```

Comment:
What's the rationale behind this specific default? Shouldn't this argument be mandatory? I can't imagine anyone calling this method without a string to format

Reply:
Because `build_docs` will fail if it is omitted entirely, and if you specify a blank value, the current docs that will be generated will not show what it is supposed to do. Do you have any suggestions?

Reply:
We should probably fix `build_docs`, but that's a much bigger project and out of scope. Let's keep this default.

Reply:
Got it! It is getting late here, so I will try to fix the errors tomorrow.